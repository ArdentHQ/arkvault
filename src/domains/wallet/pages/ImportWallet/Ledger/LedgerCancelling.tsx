import React from "react";
import { useTranslation } from "react-i18next";

import { Image } from "@/app/components/Image";
import { Spinner } from "@/app/components/Spinner";

export const LedgerCancelling = () => {
	const { t } = useTranslation();
	return (
		<div className="space-y-8" data-testid="LedgerCancellingScreen">
			<Image name="ErrorTransactionLedgerBanner" domain="transaction" className="mx-auto max-w-full" />

			<div className="inline-flex w-full items-center justify-center space-x-3">
				<Spinner />
				<span className="animate-pulse font-semibold text-theme-secondary-text">
					{t("WALLETS.PAGE_IMPORT_WALLET.CANCELLING_STATE.TITLE")}
				</span>
			</div>
		</div>
	);
};
