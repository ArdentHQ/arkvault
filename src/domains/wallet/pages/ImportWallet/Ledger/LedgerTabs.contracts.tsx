import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

import { useLedgerScanner } from "@/app/contexts";

export enum LedgerTabStep {
	ListenLedgerStep = 1,
	NetworkStep,
	LedgerConnectionStep,
	LedgerScanStep,
	LedgerImportStep,
}

export interface LedgerTabsProperties {
	activeIndex?: LedgerTabStep;
	onClickEditWalletName: (wallet: Contracts.IReadWriteWallet) => void;
}

export interface LedgerTableProperties extends ReturnType<typeof useLedgerScanner> {
	network: Networks.Network;
	isCompact: boolean;
}
