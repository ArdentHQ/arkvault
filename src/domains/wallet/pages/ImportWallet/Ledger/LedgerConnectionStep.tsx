import { Networks } from "@payvo/sdk";
import React, { useEffect, useState } from "react";
import { useFormContext } from "react-hook-form";
import { useTranslation } from "react-i18next";

import { LedgerCancelling } from "./LedgerCancelling";
import { Alert } from "@/app/components/Alert";
import { FormField, FormLabel } from "@/app/components/Form";
import { Header } from "@/app/components/Header";
import { Image } from "@/app/components/Image";
import { Spinner } from "@/app/components/Spinner";
import { useLedgerContext } from "@/app/contexts/Ledger";
import { useActiveProfile } from "@/app/hooks";
import { SelectNetwork } from "@/domains/network/components/SelectNetwork";

const ConnectionContent = ({
	error,
	isConnected,
	coinName,
}: {
	isConnected: boolean;
	error: string;
	coinName: string;
}) => {
	const { t } = useTranslation();

	if (isConnected) {
		return <Alert variant="success">{t("WALLETS.MODAL_LEDGER_WALLET.CONNECT_SUCCESS")}</Alert>;
	}

	if (error) {
		return <Alert variant="danger">{error}</Alert>;
	}

	return (
		<div className="space-y-8">
			<Image name="WaitingLedgerDevice" domain="wallet" className="mx-auto max-w-full" />

			<div className="inline-flex w-full items-center justify-center space-x-3">
				<Spinner />
				<span className="animate-pulse font-semibold text-theme-secondary-text">
					{t("WALLETS.MODAL_LEDGER_WALLET.OPEN_APP", { coin: coinName })}
				</span>
			</div>
		</div>
	);
};

export const LedgerConnectionStep = ({
	onConnect,
	onFailed,
	cancelling,
}: {
	cancelling: boolean;
	onConnect?: () => void;
	onFailed?: (error: Error) => void;
}) => {
	const { t } = useTranslation();
	const activeProfile = useActiveProfile();

	const { watch, register, setValue, unregister } = useFormContext();
	const { connect, abortConnectionRetry, error, isConnected } = useLedgerContext();

	const [network] = useState<Networks.Network>(() => watch("network"));

	useEffect(() => {
		register("connected", { required: true });

		return () => {
			unregister("connected");
		};
	}, [register, unregister]);

	useEffect(() => {
		connect(activeProfile, network.coin(), network.id());
	}, [activeProfile, network]); // eslint-disable-line react-hooks/exhaustive-deps

	useEffect(() => {
		if (error) {
			onFailed?.(new Error(error));
		}
	}, [isConnected, error]);

	useEffect(() => {
		if (isConnected) {
			setValue("connected", true, { shouldDirty: true, shouldValidate: true });
			onConnect?.();
		}
	}, [isConnected, onConnect, setValue]);

	useEffect(
		() => () => {
			abortConnectionRetry();
		},
		[abortConnectionRetry],
	);

	if (cancelling) {
		return <LedgerCancelling />;
	}

	return (
		<section data-testid="LedgerConnectionStep" className="space-y-8">
			<Header
				title={t("WALLETS.PAGE_IMPORT_WALLET.LEDGER_CONNECTION_STEP.TITLE")}
				subtitle={t("WALLETS.PAGE_IMPORT_WALLET.LEDGER_CONNECTION_STEP.SUBTITLE")}
			/>

			<FormField name="network">
				<FormLabel label={t("COMMON.CRYPTOASSET")} />
				<SelectNetwork
					id="ImportWallet__network"
					networks={[network, network, network]}
					selectedNetwork={network}
					isDisabled
					profile={activeProfile}
				/>
			</FormField>

			<ConnectionContent error={error} isConnected={isConnected} coinName={network.coin()} />
		</section>
	);
};
