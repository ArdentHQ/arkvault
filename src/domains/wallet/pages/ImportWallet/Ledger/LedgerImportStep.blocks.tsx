import React from "react";
import { Networks } from "@ardenthq/sdk";
import { Contracts } from "@ardenthq/sdk-profiles";
import { useTranslation } from "react-i18next";
import { Address } from "@/app/components/Address";
import { Amount } from "@/app/components/Amount";
import { Button } from "@/app/components/Button";
import { Icon } from "@/app/components/Icon";
import { LedgerData } from "@/app/contexts/Ledger";
import { assertWallet } from "@/utils/assertions";
import { useBreakpoint } from "@/app/hooks";
import { AmountWrapper } from "./LedgerScanStep.blocks";
import { MobileCard } from "@/app/components/Table/Mobile/MobileCard";
import { MobileSection } from "@/app/components/Table/Mobile/MobileSection";

export const SectionHeaderMobile = ({ title }: { title: string }) => (
	<div
		className="flex h-9 w-full flex-row items-center justify-between border-l-2 border-l-theme-primary-400 bg-theme-primary-100 px-3 dark:border-l-theme-primary-300 dark:bg-theme-secondary-800"
		data-testid="SectionHeaderMobile__wrapper"
	>
		<span className="text-base font-semibold text-theme-secondary-700 dark:text-theme-secondary-500">{title}</span>
	</div>
);

export const SectionBodyItem = ({ title, children }: { title: string; children: React.ReactNode }) => (
	<div
		className="flex w-full flex-row items-center justify-between text-sm font-semibold sm:justify-start sm:gap-3 sm:text-base"
		data-testid="SectionBodyItem__wrapper"
	>
		<span className="text-theme-secondary-700 dark:text-theme-secondary-500 sm:w-[90px]">{title}</span>
		<div className="text-theme-secondary-900 dark:text-theme-secondary-200 sm:w-full">{children}</div>
	</div>
);

const EditButton = ({ onClick }: { onClick: () => void }) => {
	const { t } = useTranslation();

	return (
		<Button
			variant="transparent"
			onClick={onClick}
			className="!p-0 text-theme-primary-600 dark:text-theme-secondary-500"
			data-testid="LedgerImportStep__edit-alias"
		>
			<Icon name="Pencil" />
			<span>{t("COMMON.EDIT")}</span>
		</Button>
	);
};

const DesktopImportSection = ({ title, children }: { title: string; children: React.ReactNode }) => (
	<div className="flex flex-col gap-4" data-tesid="DesktopImportSection__wrapper">
		<div className="flex flex-col gap-2">
			<span className="text-base font-semibold text-theme-secondary-700">{title}</span>
			<div className="flex flex-col gap-4 rounded-xl border border-theme-secondary-300 px-6 py-5 dark:border-theme-secondary-800">
				{children}
			</div>
		</div>
	</div>
);

export const SingleImport = ({
	network,
	onClickEditWalletName,
	profile,
	wallets,
}: {
	network: Networks.Network;
	onClickEditWalletName: (wallet: Contracts.IReadWriteWallet) => void;
	profile: Contracts.IProfile;
	wallets: LedgerData[];
}) => {
	const { t } = useTranslation();
	const { isXs } = useBreakpoint();

	const ledgerWallet = wallets[0];

	const wallet = profile.wallets().findByAddressWithNetwork(ledgerWallet.address, network.id());
	assertWallet(wallet);

	if (isXs) {
		return (
			<div className="flex flex-col gap-6" data-testid="SingleImport__container-mobile">
				<div>
					<SectionHeaderMobile title={t("COMMON.IMPORTED")} />
					<div className="flex flex-col gap-3 px-3 pt-3">
						<SectionBodyItem title={t("COMMON.ADDRESS")}>
							<div className="w-32">
								<Address address={ledgerWallet.address} showCopyButton wrapperClass="justify-between" />
							</div>
						</SectionBodyItem>

						<SectionBodyItem title={t("COMMON.BALANCE")}>
							<Amount value={ledgerWallet.balance ?? 0} ticker={network.ticker()} />
						</SectionBodyItem>
					</div>
				</div>

				<div>
					<SectionHeaderMobile title={t("COMMON.ADDRESS_NAME")} />
					<div className="flex flex-col gap-3 px-3 pt-3">
						<SectionBodyItem title={t("COMMON.NAME")}>
							<div className="flex flex-row items-center gap-2">
								{wallet.alias()}
								<hr className="h-5 w-px border-transparent bg-theme-secondary-300 dark:bg-theme-secondary-800" />
								<EditButton onClick={() => onClickEditWalletName(wallet)} />
							</div>
						</SectionBodyItem>
					</div>
				</div>
			</div>
		);
	}

	return (
		<div className="flex flex-col gap-4">
			<DesktopImportSection title={t("COMMON.IMPORTED")}>
				<SectionBodyItem title={t("COMMON.ADDRESS")}>
					<Address address={ledgerWallet.address} showCopyButton />
				</SectionBodyItem>

				<hr className="w-full border border-dashed border-theme-secondary-300 dark:border-theme-secondary-800" />

				<SectionBodyItem title={t("COMMON.BALANCE")}>
					<Amount value={ledgerWallet.balance ?? 0} ticker={network.ticker()} />
				</SectionBodyItem>
			</DesktopImportSection>

			<DesktopImportSection title={t("COMMON.ADDRESS_NAME")}>
				<SectionBodyItem title={t("COMMON.NAME")}>
					<div className="flex w-full flex-row items-center justify-between">
						<span>{wallet.alias()}</span>

						<EditButton onClick={() => onClickEditWalletName(wallet)} />
					</div>
				</SectionBodyItem>
			</DesktopImportSection>
		</div>
	);
};

export const ImportedLedgerMobileItem = ({
	address,
	balance,
	coin,
	name,
	onClick,
}: {
	address: string;
	balance?: number;
	coin: string;
	name: string;
	onClick: () => void;
}) => {
	const { t } = useTranslation();

	return (
		<MobileCard data-testid="LedgerMobileItem__wrapper">
			<div className="flex h-11 w-full items-center justify-between bg-theme-secondary-100 px-4 dark:bg-black">
				<span className="text-sm font-semibold text-theme-secondary-700 dark:text-theme-secondary-500">
					{name}
				</span>

				<EditButton onClick={onClick} />
			</div>

			<div className="flex w-full flex-col gap-4 px-4 pb-4 pt-2.5">
				<MobileSection title={t("COMMON.ADDRESS")}>
					<Address
						address={address}
						showCopyButton
						addressClass="text-theme-secondary-900 text-sm font-semibold dark:text-theme-secondary-200"
					/>
				</MobileSection>
				<MobileSection title={`${t("COMMON.VALUE")} (${coin})`} data-testid="LedgerMobileItem__network">
					<AmountWrapper isLoading={false}>
						<Amount value={balance!} ticker={coin} />
					</AmountWrapper>
				</MobileSection>
			</div>
		</MobileCard>
	);
};
