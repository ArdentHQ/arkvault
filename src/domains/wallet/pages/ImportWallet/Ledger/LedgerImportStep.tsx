import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";
import React, { useState } from "react";
import { useFormContext } from "react-hook-form";
import { useTranslation } from "react-i18next";

import { Address } from "@/app/components/Address";
import { Amount } from "@/app/components/Amount";
import { Avatar } from "@/app/components/Avatar";
import { Button } from "@/app/components/Button";
import { Header } from "@/app/components/Header";
import { Icon } from "@/app/components/Icon";
import { Tooltip } from "@/app/components/Tooltip";
import { LedgerData } from "@/app/contexts/Ledger";
import { TransactionDetail, TransactionNetwork } from "@/domains/transaction/components/TransactionDetail";
import { assertNetwork, assertWallet } from "@/utils/assertions";

const MultipleImport = ({
	network,
	onClickEditWalletName,
	profile,
	wallets,
}: {
	network: Networks.Network;
	onClickEditWalletName: (wallet: Contracts.IReadWriteWallet) => void;
	profile: Contracts.IProfile;
	wallets: LedgerData[];
}) => {
	const { t } = useTranslation();

	return (
		<div>
			<ul>
				{wallets.map((wallet) => {
					const importedWallet = profile.wallets().findByAddressWithNetwork(wallet.address, network.id());
					assertWallet(importedWallet);

					return (
						<li key={wallet.address}>
							<TransactionDetail
								className="py-4"
								paddingPosition="none"
								borderPosition="bottom"
								extra={
									<Tooltip content={t("WALLETS.WALLET_NAME")}>
										<Button
											data-testid="LedgerImportStep__edit-alias"
											type="button"
											variant="secondary"
											onClick={() => onClickEditWalletName(importedWallet)}
										>
											<Icon name="Pencil" />
										</Button>
									</Tooltip>
								}
							>
								<div className="flex w-0 flex-1 items-center space-x-3 overflow-hidden">
									<Avatar size="lg" address={wallet.address} />

									<div className="flex w-full min-w-0 flex-col justify-between">
										<Address walletName={importedWallet.alias()} address={wallet.address} />
										<p className="text-sm font-medium text-theme-secondary-500">
											<Amount value={wallet.balance ?? 0} ticker={network.ticker()} />
										</p>
									</div>
								</div>
							</TransactionDetail>
						</li>
					);
				})}
			</ul>
		</div>
	);
};

const SingleImport = ({
	network,
	onClickEditWalletName,
	profile,
	wallets,
}: {
	network: Networks.Network;
	onClickEditWalletName: (wallet: Contracts.IReadWriteWallet) => void;
	profile: Contracts.IProfile;
	wallets: LedgerData[];
}) => {
	const { t } = useTranslation();

	const ledgerWallet = wallets[0];

	const wallet = profile.wallets().findByAddressWithNetwork(ledgerWallet.address, network.id());
	assertWallet(wallet);

	return (
		<>
			<TransactionDetail
				label={t("COMMON.ADDRESS")}
				extra={<Avatar size="lg" address={ledgerWallet.address} />}
				borderPosition="bottom"
				paddingPosition="bottom"
			>
				<Address address={ledgerWallet.address} />
			</TransactionDetail>

			<TransactionDetail label={t("COMMON.BALANCE")} borderPosition="bottom" paddingPosition="bottom">
				<Amount value={ledgerWallet.balance ?? 0} ticker={network.ticker()} />
			</TransactionDetail>

			<TransactionDetail
				label={t("WALLETS.WALLET_NAME")}
				padding={false}
				border={false}
				extra={
					<Button
						data-testid="LedgerImportStep__edit-alias"
						type="button"
						variant="secondary"
						onClick={() => onClickEditWalletName(wallet)}
					>
						<Icon name="Pencil" />
					</Button>
				}
			>
				{wallet.alias()}
			</TransactionDetail>
		</>
	);
};

export const LedgerImportStep = ({
	onClickEditWalletName,
	profile,
	wallets,
}: {
	wallets: LedgerData[];
	profile: Contracts.IProfile;
	onClickEditWalletName: (wallet: Contracts.IReadWriteWallet) => void;
}) => {
	const { t } = useTranslation();

	const { watch } = useFormContext();

	const [network] = useState(() => watch("network"));
	assertNetwork(network);

	return (
		<section data-testid="LedgerImportStep" className="space-y-6">
			<Header
				title={t("WALLETS.PAGE_IMPORT_WALLET.LEDGER_IMPORT_STEP.TITLE")}
				subtitle={t("WALLETS.PAGE_IMPORT_WALLET.LEDGER_IMPORT_STEP.SUBTITLE")}
			/>

			<TransactionNetwork network={network} borderPosition="bottom" paddingPosition="bottom" />

			{wallets.length > 1 ? (
				<MultipleImport
					wallets={wallets}
					profile={profile}
					network={network}
					onClickEditWalletName={onClickEditWalletName}
				/>
			) : (
				<SingleImport
					wallets={wallets}
					profile={profile}
					network={network}
					onClickEditWalletName={onClickEditWalletName}
				/>
			)}
		</section>
	);
};
