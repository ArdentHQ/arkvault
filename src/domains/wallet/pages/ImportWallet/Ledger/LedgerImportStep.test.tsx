import Transport from "@ledgerhq/hw-transport";
import { Contracts } from "@payvo/sdk-profiles";
import userEvent from "@testing-library/user-event";
import React from "react";
import { FormProvider, useForm } from "react-hook-form";

import { LedgerImportStep } from "./LedgerImportStep";
import { LedgerData } from "@/app/contexts";
import { LedgerProvider } from "@/app/contexts/Ledger/Ledger";
import { getDefaultAlias } from "@/domains/wallet/utils/get-default-alias";
import { env, getDefaultProfileId, renderResponsive, screen, waitFor } from "@/utils/testing-library";

describe("LedgerImportStep", () => {
	let profile: Contracts.IProfile;
	let transport: typeof Transport;

	const derivationPath = "m/44'/1'/0'/0/3";

	const ledgerWallets: LedgerData[] = [
		{ address: "DJpFwW39QnQvQRQJF2MCfAoKvsX4DJ28jq", balance: 0, path: derivationPath },
		{ address: "DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES", isNew: true, path: derivationPath },
	];

	beforeEach(async () => {
		profile = env.profiles().findById(getDefaultProfileId());

		for (const { address, path } of ledgerWallets) {
			const wallet = await profile.walletFactory().fromAddressWithDerivationPath({
				address,
				coin: "ARK",
				network: "ark.devnet",
				path,
			});

			profile.wallets().push(wallet);

			wallet.mutator().alias(
				getDefaultAlias({
					network: wallet.network(),
					profile,
				}),
			);
		}

		jest.useFakeTimers();
	});

	afterEach(() => {
		jest.runOnlyPendingTimers();
		jest.useRealTimers();

		for (const { address } of ledgerWallets) {
			const wallet = profile.wallets().findByAddressWithNetwork(address, "ark.devnet");

			if (wallet) {
				profile.wallets().forget(wallet.id());
			}
		}
	});

	const renderComponent = (breakpoint: string, wallets: LedgerData[] = ledgerWallets) => {
		const onClickEditWalletName = jest.fn();

		const network = profile.wallets().findByAddressWithNetwork(wallets[0].address, "ark.devnet")?.network();

		const Component = () => {
			const form = useForm<any>({
				defaultValues: { network },
			});

			return (
				<FormProvider {...form}>
					<LedgerProvider transport={transport}>
						<LedgerImportStep
							onClickEditWalletName={onClickEditWalletName}
							wallets={wallets}
							profile={profile}
						/>
					</LedgerProvider>
				</FormProvider>
			);
		};

		return {
			...renderResponsive(<Component />, breakpoint),
			onClickEditWalletName,
		};
	};

	it.each(["xs", "lg"])("should render with single import (%s)", (breakpoint) => {
		const { container, onClickEditWalletName } = renderComponent(breakpoint, ledgerWallets.slice(1));

		userEvent.click(screen.getByTestId("LedgerImportStep__edit-alias"));

		expect(onClickEditWalletName).toHaveBeenCalledTimes(1);
		expect(container).toMatchSnapshot();
	});

	it.each(["xs", "lg"])("should render with multiple import", async (breakpoint) => {
		const { container, onClickEditWalletName } = renderComponent(breakpoint);

		await waitFor(() => expect(screen.getAllByTestId("LedgerImportStep__edit-alias")).toHaveLength(2));

		userEvent.click(screen.getAllByTestId("LedgerImportStep__edit-alias")[0]);

		expect(onClickEditWalletName).toHaveBeenCalledTimes(1);
		expect(container).toMatchSnapshot();
	});
});
