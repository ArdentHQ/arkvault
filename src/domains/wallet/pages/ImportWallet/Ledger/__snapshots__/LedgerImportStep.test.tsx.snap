// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LedgerImportStep should render with multiple import 1`] = `
<div>
  <section
    class="space-y-6"
    data-testid="LedgerImportStep"
  >
    <div
      class="flex items-end justify-between bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Completed
        </h1>
        <div
          class="flex items-center text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          Your Ledger addresses have been imported.
        </div>
      </div>
    </div>
    <div
      class="no-ligatures css-3310kx"
      data-testid="TransactionNetwork"
    >
      <div
        class="w-40 flex-1 space-y-2 whitespace-nowrap"
        data-testid="TransactionDetail--desktop"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Cryptoasset
        </div>
        <div
          class="flex items-center font-semibold"
        >
          ARK Devnet
        </div>
      </div>
      <div
        aria-label="ARK Devnet"
        class="text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700 ring-theme-background duration-100 css-14jhlj1"
        data-testid="NetworkIcon-ARK-ark.devnet"
      >
        <div
          class="css-1i4b0eq"
          data-testid="NetworkIcon__icon"
          height="20"
          width="20"
        >
          <svg>
            ark.svg
          </svg>
        </div>
      </div>
    </div>
    <div>
      <ul>
        <li>
          <div
            class="no-ligatures py-4 css-1s6a8s3"
            data-testid="TransactionDetail"
          >
            <div
              class="w-40 flex-1 space-y-2 whitespace-nowrap"
              data-testid="TransactionDetail--desktop"
            >
              <div
                class="flex items-center font-semibold"
              >
                <div
                  class="flex w-0 flex-1 items-center space-x-3 overflow-hidden"
                >
                  <div
                    class="shrink-0 css-8d0u1i"
                    data-testid="Avatar"
                  >
                    <div
                      class="inline-flex h-full w-full items-center justify-center overflow-hidden rounded-full align-middle"
                    >
                      <img
                        alt="DJpFwW39QnQvQRQJF2MCfAoKvsX4DJ28jq"
                        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(103, 58, 183)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"55\\" cx=\\"60\\" cy=\\"70\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"35\\" cx=\\"30\\" cy=\\"50\\" fill=\\"rgb(139, 195, 74)\\"/><circle r=\\"60\\" cx=\\"20\\" cy=\\"60\\" fill=\\"rgb(255, 152, 0)\\"/></svg>"
                        title="DJpFwW39QnQvQRQJF2MCfAoKvsX4DJ28jq"
                      />
                    </div>
                  </div>
                  <div
                    class="flex w-full min-w-0 flex-col justify-between"
                  >
                    <div
                      class="no-ligatures flex overflow-hidden whitespace-nowrap items-center space-x-2 w-full"
                    >
                      <span
                        class="font-semibold text-base text-theme-text"
                        data-testid="Address__alias"
                      >
                        <span
                          class="css-u3jrna"
                          data-testid="TruncateEnd"
                        >
                          ARK Devnet #3
                        </span>
                      </span>
                      <span
                        class="min-w-0 overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                        data-testid="Address__address"
                      >
                        DJpFwW39QnQvQRQJF2MCfAoKvsX4DJ28jq
                      </span>
                    </div>
                    <p
                      class="text-sm font-medium text-theme-secondary-500"
                    >
                      <span
                        class="whitespace-nowrap"
                        data-testid="Amount"
                      >
                        0 DARK
                      </span>
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="css-1jhy6v9"
              data-testid="LedgerImportStep__edit-alias"
              type="button"
            >
              <div
                class="flex items-center space-x-2"
              >
                <div
                  class="css-v0ob3f"
                  height="16"
                  width="16"
                >
                  <svg>
                    pencil.svg
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </li>
        <li>
          <div
            class="no-ligatures py-4 css-1s6a8s3"
            data-testid="TransactionDetail"
          >
            <div
              class="w-40 flex-1 space-y-2 whitespace-nowrap"
              data-testid="TransactionDetail--desktop"
            >
              <div
                class="flex items-center font-semibold"
              >
                <div
                  class="flex w-0 flex-1 items-center space-x-3 overflow-hidden"
                >
                  <div
                    class="shrink-0 css-8d0u1i"
                    data-testid="Avatar"
                  >
                    <div
                      class="inline-flex h-full w-full items-center justify-center overflow-hidden rounded-full align-middle"
                    >
                      <img
                        alt="DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES"
                        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(63, 81, 181)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"40\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/><circle r=\\"55\\" cx=\\"60\\" cy=\\"60\\" fill=\\"rgb(205, 220, 57)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(33, 150, 243)\\"/></svg>"
                        title="DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES"
                      />
                    </div>
                  </div>
                  <div
                    class="flex w-full min-w-0 flex-col justify-between"
                  >
                    <div
                      class="no-ligatures flex overflow-hidden whitespace-nowrap items-center space-x-2 w-full"
                    >
                      <span
                        class="font-semibold text-base text-theme-text"
                        data-testid="Address__alias"
                      >
                        <span
                          class="css-u3jrna"
                          data-testid="TruncateEnd"
                        >
                          ARK Devnet #4
                        </span>
                      </span>
                      <span
                        class="min-w-0 overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                        data-testid="Address__address"
                      >
                        DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES
                      </span>
                    </div>
                    <p
                      class="text-sm font-medium text-theme-secondary-500"
                    >
                      <span
                        class="whitespace-nowrap"
                        data-testid="Amount"
                      >
                        0 DARK
                      </span>
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="css-1jhy6v9"
              data-testid="LedgerImportStep__edit-alias"
              type="button"
            >
              <div
                class="flex items-center space-x-2"
              >
                <div
                  class="css-v0ob3f"
                  height="16"
                  width="16"
                >
                  <svg>
                    pencil.svg
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </li>
      </ul>
    </div>
  </section>
</div>
`;

exports[`LedgerImportStep should render with single import 1`] = `
<div>
  <section
    class="space-y-6"
    data-testid="LedgerImportStep"
  >
    <div
      class="flex items-end justify-between bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Completed
        </h1>
        <div
          class="flex items-center text-theme-secondary-text"
          data-testid="header__subtitle"
        >
          Your Ledger addresses have been imported.
        </div>
      </div>
    </div>
    <div
      class="no-ligatures css-3310kx"
      data-testid="TransactionNetwork"
    >
      <div
        class="w-40 flex-1 space-y-2 whitespace-nowrap"
        data-testid="TransactionDetail--desktop"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Cryptoasset
        </div>
        <div
          class="flex items-center font-semibold"
        >
          ARK Devnet
        </div>
      </div>
      <div
        aria-label="ARK Devnet"
        class="text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700 ring-theme-background duration-100 css-14jhlj1"
        data-testid="NetworkIcon-ARK-ark.devnet"
      >
        <div
          class="css-1i4b0eq"
          data-testid="NetworkIcon__icon"
          height="20"
          width="20"
        >
          <svg>
            ark.svg
          </svg>
        </div>
      </div>
    </div>
    <div
      class="no-ligatures css-3310kx"
      data-testid="TransactionDetail"
    >
      <div
        class="w-40 flex-1 space-y-2 whitespace-nowrap"
        data-testid="TransactionDetail--desktop"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Address
        </div>
        <div
          class="flex items-center font-semibold"
        >
          <div
            class="no-ligatures flex overflow-hidden whitespace-nowrap items-center space-x-2 w-full"
          >
            <span
              class="min-w-0 overflow-hidden text-theme-text font-semibold text-base"
              data-testid="Address__address"
            >
              DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES
            </span>
          </div>
        </div>
      </div>
      <div
        class="shrink-0 css-8d0u1i"
        data-testid="Avatar"
      >
        <div
          class="inline-flex h-full w-full items-center justify-center overflow-hidden rounded-full align-middle"
        >
          <img
            alt="DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES"
            src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(63, 81, 181)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"40\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(0, 150, 136)\\"/><circle r=\\"55\\" cx=\\"60\\" cy=\\"60\\" fill=\\"rgb(205, 220, 57)\\"/><circle r=\\"50\\" cx=\\"50\\" cy=\\"30\\" fill=\\"rgb(33, 150, 243)\\"/></svg>"
            title="DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES"
          />
        </div>
      </div>
    </div>
    <div
      class="no-ligatures css-3310kx"
      data-testid="TransactionDetail"
    >
      <div
        class="w-40 flex-1 space-y-2 whitespace-nowrap"
        data-testid="TransactionDetail--desktop"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Balance
        </div>
        <div
          class="flex items-center font-semibold"
        >
          <span
            class="whitespace-nowrap"
            data-testid="Amount"
          >
            0 DARK
          </span>
        </div>
      </div>
    </div>
    <div
      class="no-ligatures css-x92ash"
      data-testid="TransactionDetail"
    >
      <div
        class="w-40 flex-1 space-y-2 whitespace-nowrap"
        data-testid="TransactionDetail--desktop"
      >
        <div
          class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Wallet Name
        </div>
        <div
          class="flex items-center font-semibold"
        >
          ARK Devnet #4
        </div>
      </div>
      <button
        class="css-1jhy6v9"
        data-testid="LedgerImportStep__edit-alias"
        type="button"
      >
        <div
          class="flex items-center space-x-2"
        >
          <div
            class="css-v0ob3f"
            height="16"
            width="16"
          >
            <svg>
              pencil.svg
            </svg>
          </div>
        </div>
      </button>
    </div>
  </section>
</div>
`;
